front/src/common/Map/Search/MapSearchSignal.tsx

```diff
@@ -0,0 +1,258 @@
+import { useDebounce } from 'utils/helpers';
+import { useTranslation } from 'react-i18next';
+import { getMap } from 'reducers/map/selectors';
+import { AllGeoJSON } from '@turf/helpers';
+import MultiSelectSNCF from 'common/BootstrapSNCF/MultiSelectSNCF';
+import { osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import { getInfraID } from 'reducers/osrdconf/selectors';
+import { ISignalSearchResult } from './searchTypes';
+import { searchPayloadType, signalAspects } from '../const';
+import SignalCard from './SignalCard';
+
+type MapSearchSignalProps = {
+  updateExtViewport: (viewport: Partial<Viewport>) => void;
+};
+
+export type SortType = {
+  name: 'label' | 'type' | 'line_name' | 'line_code';
+  order: number;
+};
+
+const SIGNAL_ASPECTS = signalAspects;
+
+const MapSearchSignal = ({ updateExtViewport }: MapSearchSignalProps) => {
+  const map = useSelector(getMap);
+  const infraID = useSelector(getInfraID);
+
+  const [searchState, setSearch] = useState('');
+  const [searchLineState, setSearchLine] = useState('');
+  const [aspect, setAspect] = useState<string[]>([]);
+  const [searchResults, setSearchResults] = useState<ISignalSearchResult[]>([]);
+  // Sort by, and order 0 = ASC, 1 = DESC
+  const [sortFilter, setSortFilter] = useState<SortType>({
+    name: 'label',
+    order: 0,
+  });
+  const dispatch = useDispatch();
+  const [postSearch] = osrdEditoastApi.usePostSearchMutation();
+  const { t } = useTranslation(['translation', 'map-search']);
+
+  const updateSearch = async (params: searchPayloadType) => {
+    await postSearch({
+      body: { object: params.object, query: params.query },
+    })
+      .unwrap()
+      .then((results) => {
+        setSearchResults(results as ISignalSearchResult[]);
+      })
+      .catch((error) => {
+        console.error(error);
+        setSearchResults([]);
+      });
+  };
+
+  const debouncedSearchTerm = useDebounce(searchState, 300);
+  const debouncedSearchLine = useDebounce(searchLineState, 300);
+
+  const getPayload = () => {
+    let playloadQuery: (string | number | string[])[] = ['search', ['line_name'], searchState];
+
+    if (searchLineState) playloadQuery = ['=', ['line_code'], Number(searchLineState)];
+
+    const aspectQuery = aspect.join(' ');
+    const payload = {
+      object: 'signal',
+      query: [
+        'and',
+        ['=', ['infra_id'], infraID],
+        playloadQuery,
+        !aspectQuery || ['contains', ['list', aspectQuery], ['aspects']],
+      ],
+    };
+
+    return payload;
+  };
+
+  useEffect(() => {
+    if (searchState || searchLineState) {
+      updateSearch(getPayload());
+    }
+    if (!searchState && !searchLineState) {
+      setSearchResults([]);
+    }
+  }, [debouncedSearchTerm, debouncedSearchLine, aspect]);
+
+  const onResultClick = (result: ISignalSearchResult) => {
+    const coordinates = map.mapTrackSources === 'schematic' ? result.schematic : result.geographic;
```

Comment:
Can you create a function getCoordinates for instance ?
It would be nice to create a file src/common/map/utils.ts (you can put the function in the this new file :))

Reply:
Done